### PR TITLE
GLTFExporter: Export textures as data URIs.

### PR DESCRIPTION
**Pull request by donmccurdy**
/cc @fernandojsg

### CODE REVIEW

**Comment by donmccurdy** [comment]
Maybe there should be a way to opt out of this? But I think data URIs would be a better default than remote URLs.

File: examples/js/exporters/GLTFExporter.js [hunk 1]
```diff
@@ -370,14 +370,20 @@ THREE.GLTFExporter.prototype = {
 
 			var gltfImage = {};
 
+			var canvas = document.createElement( 'canvas' );
```
Comment (fernandojsg) [inline examples/js/exporters/GLTFExporter.js]: What about creating just one canvas and reuse it on every image by resizing it?

File: examples/js/exporters/GLTFExporter.js [hunk 2]
```diff
@@ -370,14 +370,20 @@ THREE.GLTFExporter.prototype = {
...
+
 			if ( options.embedImages ) {
 
 				// @TODO { bufferView, mimeType }
 
 			} else {
 
-				// @TODO base64 based on options
-				gltfImage.uri = map.image.src;
+				gltfImage.uri = canvas.toDataURL( format );
```
Comment (fernandojsg) [inline examples/js/exporters/GLTFExporter.js]: I also think including a opt could be nice to force URI (as default) or just using the external URL.

**Review by fernandojsg — COMMENTED** [review]
Looking cool, I would add the opt even if default should be URI yep

File: examples/js/exporters/GLTFExporter.js (hunk 1 above)
Comment (donmccurdy) [reply examples/js/exporters/GLTFExporter.js]: Done.

File: examples/js/exporters/GLTFExporter.js (hunk 2 above)
Comment (donmccurdy) [reply examples/js/exporters/GLTFExporter.js]: Done.

**Comment by donmccurdy** [comment]
^Made those changes and replied "Done" but now my WiFi isn't letting me `git push`... one more commit coming here... 😕

**Comment by fernandojsg** [comment]
You code faster than the internet!!!

**Comment by donmccurdy** [comment]
Ok, should be all set now! 😅

**Comment by fernandojsg** [comment]
r+

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**